rest/src/main/java/eu/europa/ec/fisheries/uvms/activity/rest/resources/FishingActivityResource.java

```diff
@@ -129,13 +129,12 @@ public Response getAllCorrections(@Context HttpServletRequest request,
     public Response getPreviousFishingActivity(@Context HttpServletRequest request,
                                                @Context HttpServletResponse response,
                                                @PathParam("activityId") String activityId) {
-        int converstedActivityId = 0;
-        log.debug("Received ActivityId from frontEnd as: " + activityId);
+        int activityIdInteger = 0;
+        log.debug("Received Activity ID {}", activityId);
         if (activityId != null) {
```

Comment:
I prefer a call to Integer.parseInt() inside a try-catch, and if the parsing fails we return a HTTP 400, instead of searching for activity with ID "0".